### ISSUE TITLE
java.lang.IllegalAccessError in appengine:run

### ISSUE THREAD

**Issue by katafractari** [GoogleCloudPlatform/app-maven-plugin#205]
I'm using appengine-maven-plugin from com.google.cloud.tools for my Endpoints+Datastore app running in Appengine Standard environement. Deploying to appengine works without a problem, but I am unable to run the app locally with appengine:run.

I've attached my log and the list of dependencies used in the project.

```
[INFO] --- appengine-maven-plugin:1.3.1:run (default-cli) @ app ---
Aug 23, 2017 11:19:59 AM com.google.cloud.tools.appengine.cloudsdk.CloudSdk logCommand
INFO: submitting command: /usr/lib/jvm/java-8-openjdk/jre/bin/java -Xms512m -Xmx512m -XX:PermSize=128m -XX:MaxPermSize=256m -agentlib:jdwp=transport=dt_socket,address=8081,server=y,suspend=n -Duse_jetty9_runtime=true -D--enable_all_permissions=true -Dappengine.sdk.root=/opt/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java -cp /opt/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/lib/appengine-tools-api.jar com.google.appengine.tools.development.DevAppServerMain --allow_remote_shutdown --disable_update_check --no_java_agent /home/user/workspace/app-api/target/app-15-0-1
[INFO] GCLOUD: OpenJDK 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
[INFO] GCLOUD: OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
[INFO] GCLOUD: Listening for transport dt_socket at address: 8081
[INFO] GCLOUD: Aug 23, 2017 11:20:00 AM com.google.appengine.tools.development.SystemPropertiesManager setSystemProperties
[INFO] GCLOUD: INFO: Overwriting system property key 'java.util.logging.config.file', value '/opt/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/config/sdk/logging.properties' with value 'WEB-INF/logging.properties' from '/home/user/workspace/app-api/target/app-15-0-1/WEB-INF/appengine-web.xml'
[INFO] GCLOUD: 2017-08-23 11:20:00.416:INFO::main: Logging initialized @357ms
[INFO] GCLOUD: Aug 23, 2017 9:20:00 AM com.google.appengine.tools.development.IsolatedAppClassLoader checkWorkingDirectory
[INFO] GCLOUD: WARNING: Your working directory, (/home/user/workspace/app-api) is not equal to your 
[INFO] GCLOUD: web application root (/home/user/workspace/app-api/target/app-15-0-1)
[INFO] GCLOUD: You will not be able to access files from your working directory on the production server.
[INFO] GCLOUD: 
[INFO] GCLOUD: 2017-08-23 11:20:00.793:INFO:oejs.Server:main: jetty-9.3.18.v20170406
[INFO] GCLOUD: 2017-08-23 11:20:04.024:INFO:oeja.AnnotationConfiguration:main: Scanning elapsed time=3060ms
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.app.ObjectifyListener contextInitialized
[INFO] GCLOUD: INFO: Preparing development database with default values
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM org.hibernate.validator.util.Version <clinit>
[INFO] GCLOUD: INFO: Hibernate Validator 4.2.0.Final
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.google.appengine.api.datastore.dev.LocalDatastoreService init
[INFO] GCLOUD: INFO: Local Datastore initialized: 
[INFO] GCLOUD:  Type: High Replication
[INFO] GCLOUD:  Storage: /home/user/workspace/app-api/target/app-15-0-1/WEB-INF/appengine-generated/local_db.bin
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.google.appengine.api.datastore.dev.LocalDatastoreService load
[INFO] GCLOUD: INFO: The backing store, /home/user/workspace/app-api/target/app-15-0-1/WEB-INF/appengine-generated/local_db.bin, does not exist. It will be created.
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.app.dao.UserDAO cacheByUsername
[INFO] GCLOUD: FINE: Caching user(user@app.com) took 30 ms
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.app.dao.UserDAO cacheByUsername
[INFO] GCLOUD: FINE: Caching user(example@example.com) took 1 ms
[INFO] GCLOUD: Aug 23, 2017 9:20:05 AM com.app.dao.UserDAO cacheByUsername
[INFO] GCLOUD: FINE: Caching user(receptionist@example.com) took 0 ms
[INFO] GCLOUD: Aug 23, 2017 9:20:06 AM com.app.ObjectifyListener prepareDevDB
[INFO] GCLOUD: INFO: Local Datastore prepared with DEV init values
[INFO] GCLOUD: Aug 23, 2017 9:20:06 AM com.google.api.server.spi.SystemServiceServlet init
[INFO] GCLOUD: INFO: SPI restricted: true
[INFO] GCLOUD: 2017-08-23 11:20:06.418:INFO:oejs.AbstractConnector:main: Started NetworkTrafficSelectChannelConnector@443a53df{HTTP/1.1,[http/1.1]}{localhost:8080}
[INFO] GCLOUD: Exception in thread "main" java.lang.IllegalAccessError: class com.google.appengine.repackaged.com.google.common.collect.ArrayListMultimapGwtSerializationDependencies cannot access its superclass com.google.appengine.repackaged.com.google.common.collect.AbstractListMultimap
[INFO] GCLOUD:  at java.lang.ClassLoader.defineClass1(Native Method)
[INFO] GCLOUD:  at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
[INFO] GCLOUD:  at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[INFO] GCLOUD:  at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
[INFO] GCLOUD:  at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
[INFO] GCLOUD:  at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
[INFO] GCLOUD:  at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
[INFO] GCLOUD:  at java.security.AccessController.doPrivileged(Native Method)
[INFO] GCLOUD:  at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
[INFO] GCLOUD:  at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[INFO] GCLOUD:  at com.google.appengine.tools.development.IsolatedAppClassLoader.loadClass(IsolatedAppClassLoader.java:195)
[INFO] GCLOUD:  at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[INFO] GCLOUD:  at java.lang.ClassLoader.defineClass1(Native Method)
[INFO] GCLOUD:  at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
[INFO] GCLOUD:  at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[INFO] GCLOUD:  at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
[INFO] GCLOUD:  at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
[INFO] GCLOUD:  at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
[INFO] GCLOUD:  at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
[INFO] GCLOUD:  at java.security.AccessController.doPrivileged(Native Method)
[INFO] GCLOUD:  at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
[INFO] GCLOUD:  at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[INFO] GCLOUD:  at com.google.appengine.tools.development.IsolatedAppClassLoader.loadClass(IsolatedAppClassLoader.java:195)
[INFO] GCLOUD:  at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[INFO] GCLOUD:  at com.google.api.server.spi.SystemService.<init>(SystemService.java:128)
[INFO] GCLOUD:  at com.google.api.server.spi.SystemServiceServlet.createSystemService(SystemServiceServlet.java:170)
[INFO] GCLOUD:  at com.google.api.server.spi.SystemServiceServlet.createSystemService(SystemServiceServlet.java:136)
[INFO] GCLOUD:  at com.google.api.server.spi.SystemServiceServlet.init(SystemServiceServlet.java:59)
[INFO] GCLOUD:  at com.google.inject.servlet.ServletDefinition.init(ServletDefinition.java:120)
[INFO] GCLOUD:  at com.google.inject.servlet.ManagedServletPipeline.init(ManagedServletPipeline.java:82)
[INFO] GCLOUD:  at com.google.inject.servlet.ManagedFilterPipeline.initPipeline(ManagedFilterPipeline.java:103)
[INFO] GCLOUD:  at com.google.inject.servlet.GuiceFilter.init(GuiceFilter.java:226)
[INFO] GCLOUD:  at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
[INFO] GCLOUD:  at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:873)
[INFO] GCLOUD:  at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:349)
[INFO] GCLOUD:  at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1406)
[INFO] GCLOUD:  at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1368)
[INFO] GCLOUD:  at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:778)
[INFO] GCLOUD:  at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:262)
[INFO] GCLOUD:  at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:522)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:105)
[INFO] GCLOUD:  at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
[INFO] GCLOUD:  at org.eclipse.jetty.server.Server.start(Server.java:422)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:105)
[INFO] GCLOUD:  at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
[INFO] GCLOUD:  at org.eclipse.jetty.server.Server.doStart(Server.java:389)
[INFO] GCLOUD:  at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[INFO] GCLOUD:  at com.google.appengine.tools.development.jetty9.JettyContainerService.startContainer(JettyContainerService.java:346)
[INFO] GCLOUD:  at com.google.appengine.tools.development.AbstractContainerService.startup(AbstractContainerService.java:284)
[INFO] GCLOUD:  at com.google.appengine.tools.development.ManualInstanceHolder.startUp(ManualInstanceHolder.java:77)
[INFO] GCLOUD:  at com.google.appengine.tools.development.AbstractModule.startup(AbstractModule.java:87)
[INFO] GCLOUD:  at com.google.appengine.tools.development.Modules.startup(Modules.java:105)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerImpl.doStart(DevAppServerImpl.java:271)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerImpl.access$000(DevAppServerImpl.java:47)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerImpl$1.run(DevAppServerImpl.java:219)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerImpl$1.run(DevAppServerImpl.java:217)
[INFO] GCLOUD:  at java.security.AccessController.doPrivileged(Native Method)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerImpl.start(DevAppServerImpl.java:217)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerMain$StartAction.apply(DevAppServerMain.java:359)
[INFO] GCLOUD:  at com.google.appengine.tools.util.Parser$ParseResult.applyArgs(Parser.java:47)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerMain.run(DevAppServerMain.java:223)
[INFO] GCLOUD:  at com.google.appengine.tools.development.DevAppServerMain.main(DevAppServerMain.java:214)
[INFO] GCLOUD: Aug 23, 2017 11:20:35 AM com.google.appengine.api.datastore.dev.LocalDatastoreService$11 run
[INFO] GCLOUD: INFO: Time to persist datastore: 5 ms
```

```
[INFO] Scanning for projects...
[WARNING] The project com.app:app:war:15-0-1 uses prerequisites which is only intended for maven-plugin projects but not for non maven-plugin projects. For such purposes you should use the maven-enforcer-plugin. See https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building app 15-0-1
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ app ---
[INFO] com.app:app:war:15-0-1
[INFO] +- com.google.appengine:appengine-api-1.0-sdk:jar:1.9.54:compile
[INFO] +- com.google.appengine:appengine-endpoints:jar:1.9.54:compile
[INFO] +- javax.servlet:servlet-api:jar:2.5:provided
[INFO] +- javax.inject:javax.inject:jar:1:compile
[INFO] +- com.googlecode.objectify:objectify:jar:5.1.21:compile
[INFO] |  \- com.google.guava:guava:jar:20.0:compile
[INFO] +- com.sappenin.objectify:objectify-utils:jar:5.1.3:compile
[INFO] +- org.hibernate:hibernate-validator:jar:4.2.0.Final:compile
[INFO] |  \- javax.validation:validation-api:jar:1.0.0.GA:compile
[INFO] +- com.google.inject.extensions:guice-servlet:jar:4.1.0:compile
[INFO] +- com.google.inject:guice:jar:4.1.0:compile
[INFO] |  \- aopalliance:aopalliance:jar:1.0:compile
[INFO] +- com.google.code.gson:gson:jar:2.8.1:compile
[INFO] +- com.google.firebase:firebase-admin:jar:5.2.0:compile
[INFO] |  +- com.google.api-client:google-api-client:jar:1.22.0:compile
[INFO] |  |  +- com.google.oauth-client:google-oauth-client:jar:1.22.0:compile
[INFO] |  |  \- com.google.http-client:google-http-client-jackson2:jar:1.22.0:compile
[INFO] |  +- com.google.api-client:google-api-client-gson:jar:1.22.0:compile
[INFO] |  +- com.google.http-client:google-http-client:jar:1.22.0:compile
[INFO] |  |  \- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO] |  \- org.json:json:jar:20160810:compile
[INFO] +- com.google.http-client:google-http-client-appengine:jar:1.22.0:compile
[INFO] +- com.google.http-client:google-http-client-gson:jar:1.22.0:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.21:compile
[INFO] +- org.slf4j:slf4j-jdk14:jar:1.7.21:compile
[INFO] +- org.slf4j:log4j-over-slf4j:jar:1.7.21:compile
[INFO] +- com.googlecode.libphonenumber:libphonenumber:jar:8.7.0:compile
[INFO] +- com.googlecode.libphonenumber:prefixmapper:jar:2.75:compile
[INFO] +- org.bitbucket.b_c:jose4j:jar:0.6.0:compile
[INFO] +- com.chargebee:chargebee-java:jar:2.3.3:compile
[INFO] +- com.sparkpost:sparkpost-lib:jar:0.18:compile
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |  |  \- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.4:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  +- com.google.code.findbugs:annotations:jar:3.0.0:compile
[INFO] |  \- com.yepher.jsondoc:annotations:jar:0.0.2:compile
[INFO] +- joda-time:joda-time:jar:2.9.9:compile
[INFO] +- org.joda:joda-money:jar:0.12:compile
[INFO] +- net.sf.supercsv:super-csv:jar:2.4.0:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.mockito:mockito-all:jar:1.10.19:test
[INFO] +- com.google.appengine:appengine-testing:jar:1.9.54:test
[INFO] +- com.google.appengine:appengine-api-stubs:jar:1.9.54:test
[INFO] +- javax.el:javax.el-api:jar:3.0.0:test
[INFO] +- org.hamcrest:hamcrest-all:jar:1.3:test
[INFO] \- com.twilio.sdk:twilio:jar:7.14.0:compile
[INFO]    +- io.jsonwebtoken:jjwt:jar:0.4:compile
[INFO]    +- org.apache.httpcomponents:httpcore:jar:4.4.4:compile
[INFO]    +- com.fasterxml.jackson.core:jackson-core:jar:2.8.7:compile
[INFO]    +- com.fasterxml.jackson.core:jackson-annotations:jar:2.8.7:compile
[INFO]    +- com.fasterxml.jackson.core:jackson-databind:jar:2.8.7:compile
[INFO]    \- javax.xml.bind:jaxb-api:jar:2.2:compile
[INFO]       +- javax.xml.stream:stax-api:jar:1.0-2:compile
[INFO]       \- javax.activation:activation:jar:1.1:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.526 s
[INFO] Finished at: 2017-08-23T11:31:33+02:00
[INFO] Final Memory: 17M/300M
[INFO] ------------------------------------------------------------------------
```

**Comment by patflynn** [comment]
@katafractari thank you for the report!

@ludoch could this be another devappserver sandbox issue?

**Comment by chanseokoh** [comment]
@katafractari can you check if this is https://issuetracker.google.com/issues/63123716? There is a workaround you can use to test this in that issue.

**Comment by katafractari** [comment]
@chanseokoh Thank you for this. I renamed appengine-remote-api.jar and now the exception is gone. Even if it's just a workaround the main thing is that I am now able to get some work done.

By the way, I'm on Google Cloud SDK 167.0.0.

**Comment by ludoch** [comment]
Can you reproduce with today's release of the Cloud SDK?

**Comment by katafractari** [comment]
@ludoch I can confirm that version 170 of the cloud SDK fixes the problem.

Here are the details of my currently installed cloud SDK components:
```
$ gcloud --version                                                                                                                        
Google Cloud SDK 170.0.1
app-engine-java 1.9.54
app-engine-python 1.9.57
bq 2.0.25
core 2017.09.08
gsutil 4.27
```

**Comment by ludoch** [comment]
Thank you!

**Comment by ludoch** [comment]
but maybe you should have:
gcloud --version  
Google Cloud SDK 171.0.0
app-engine-java 1.9.55
app-engine-python 1.9.60
bq 2.0.25
core 2017.09.11
gcloud 
gsutil 4.27


app-engine-java 1.9.55 should have the fix...

**Comment by katafractari** [comment]
@ludoch You are right. I did not rename appengine-remote-api.jar.bkp back to appengine-remote-api.jar, thus the workaround was still in place when I ran appengine:run with Google Cloud SDK 171.0.0 and app-engine-java 1.9.54. 

I properly renamed appengine-remote-api.jar.bkp back to appengine-remote-api.jar and did another upgrade of Google Cloud SDK and the appengine components:
```
Google Cloud SDK 171.0.0
app-engine-java 1.9.55
app-engine-python 1.9.60
bq 2.0.25
core 2017.09.11
gsutil 4.27
```
Now I can really confirm that the fix in app-engine-java 1.9.55 is working.
